lua/sg/components/layout/hover.lua

```diff
@@ -102,14 +102,20 @@ end
 function CodyHover:request_completion(code_only)
   self:render()
 
-  self.state:complete(self.history.bufnr, self.history.win, function(noti)
-    local lines = vim.split(noti.text, "\n")
-    if code_only then
-      table.remove(lines, 1)
-      table.remove(lines)
+  self.state:complete(self.history.bufnr, self.history.win, code_only, function(msg)
+    local lines = vim.split(msg.text, "\n")
+    local render_lines = {}
+    for _, line in ipairs(lines) do
+      if code_only then
+        if vim.trim(line) == "```" then
```

Comment:
should this match against the line? Does it ever send like "```lua" ?

Reply:
Cody's prompt gets prefilled with "```lua" so it's omitted from the start of the LLM's response. And the assumption I'm making here is that a line that's only "```" will be the end of the code response.

But this isn't perfect. I think using some xml tag shenanigans for code blocks would work better, but would require updating agent again.